torch_frame/data/mapper.py

```diff
@@ -252,3 +252,66 @@ def forward(
 
     def backward(self, tensor: Tensor) -> pd.Series:
         raise NotImplementedError
+
+
+class TextTokenizationTensorMapper(TensorMapper):
+    r"""Tokenize any text series into a dictionary of
+    :class:`MultiNestedTensor`.
+
+    Args:
+        text_tokenizer (callable): A callable function that takes list of
+            strings and returns list of dictionary. The keys of the dictionary
+            are arguments that will be put to the model, including
+            :obj:`input_ids` and :obj:`attention_mask`. The values of the
+            dictionary are tensors corresponding to keys.
+        batch_size (int, optional): The mini-batch size used for the text
+            tokenizer. If :obj:`None`, we will encode all text in a full-batch
+            manner.
+    """
+    def __init__(
+        self,
+        text_tokenizer: Callable[[List[str]], List[Dict[str, Tensor]]],
+        batch_size: Optional[int],
+    ):
+        super().__init__()
+        self.text_tokenizer = text_tokenizer
+        self.batch_size = batch_size
+
+    def forward(
+        self,
+        ser: Series,
+        *,
+        device: Optional[torch.device] = None,
+    ) -> Dict[str, MultiNestedTensor]:
+        ser = ser.astype(str)
+        ser_list = ser.tolist()
+
+        seq_dict = {}
+        if self.batch_size is None:
+            list_dict: List[Dict[str, Tensor]] = self.text_tokenizer(ser_list)
+            for key in list_dict[0]:
+                xs = [[item[key]] for item in list_dict]
+                seq_dict[key] = MultiNestedTensor.from_tensor_mat(xs)
+            for key, value in seq_dict.items():
+                seq_dict[key] = value.to(device)
+            return seq_dict
+
+        for i in tqdm(range(0, len(ser_list), self.batch_size),
+                      desc="Tokenizing texts in mini-batch"):
+            list_batch_dict: List[Dict[str, Tensor]] = self.text_tokenizer(
+                ser_list[i:i + self.batch_size])
+            batch_tokens_dict: Dict[str, MultiNestedTensor] = {}
+            for key in list_batch_dict[0]:
+                xs = [[item[key]] for item in list_batch_dict]
+                batch_tokens_dict[key] = MultiNestedTensor.from_tensor_mat(xs)
+            for key, value in batch_tokens_dict.items():
+                if key not in seq_dict:
+                    seq_dict[key] = [value]
+                else:
+                    seq_dict[key].append(value)
+        for key, values in seq_dict.items():
+            seq_dict[key] = MultiNestedTensor.cat(values, dim=0).to(device)
```

Comment:
I feel these logics can be cleaner. For instance, you don't need intermediate `batch_tokens_dict`. Also, `MultiNetedTensor.from_tensor_mat` should be only called once at the end.

Reply:
Personally, I found the naming of `list_batch_dict` unintuitive. Can have improved naming